examples/docker/modin-hdk/nyc-taxi-hdk.py

```diff
@@ -179,25 +220,66 @@ def q4_hdk(df):
         .size()
         .reset_index()
         .sort_values(
-            by=["pickup_datetime", 0], ignore_index=True, ascending=[True, False]
+            by=["pickup_datetime", 0, "passenger_count", "trip_distance"],
+            ignore_index=True, ascending=[True, False, True, True]
         )
     )
     q4_pandas_output.shape  # to trigger real execution
     return q4_pandas_output
 
 
+def q4_sql(df):
+    sql = """
+    SELECT
+        passenger_count,
+        pickup_datetime,
+        CAST(trip_distance AS int) AS trip_distance,
+        COUNT(*) AS the_count
+    FROM trips
+    GROUP BY
+        passenger_count,
+        pickup_datetime,
+        trip_distance
+    ORDER BY
+        pickup_datetime,
+        the_count desc,
+        passenger_count,
+        trip_distance
+    """
+    df["pickup_datetime"] = df["pickup_datetime"].dt.year
+    df["trip_distance"] = df["trip_distance"].astype("int64")
+    return query(sql, trips=df)
+
+
+def validate(df, hdk_func, sql_func, reset_index=True):
+    hdk_result = hdk_func(df.copy())
+    sql_result = sql_func(df.copy())
+    if reset_index:
+        hdk_result = hdk_result.reset_index()
+    hdk_result.columns = sql_result.columns
```

Comment:
Why do we to copy columns?

Reply:
Some column names are different in sql_result:
HDK cols: Index(['passenger_count', 'pickup_datetime', 0])
SQL cols: Index(['passenger_count', 'pickup_datetime', 'count'])